nixos/modules/services/mail/dovecot.nix

```diff
@@ -631,30 +587,118 @@ in
     };
 
     services.dovecot2 = {
-      protocols =
-        optional cfg.enableImap "imap" ++ optional cfg.enablePop3 "pop3" ++ optional cfg.enableLmtp "lmtp";
+      settings =
+        {
+          base_dir = mkDefault baseDir;
+          sendmail_path = mkDefault "/run/wrappers/bin/sendmail";
+          mail_plugin_dir = mkDefault "/run/current-system/sw/lib/dovecot/modules";
+          mail_path = mkDefault "maildir:/var/spool/mail/%u";
+          mail_uid = mkIf (cfg.mailUser != null) (mkDefault cfg.mailUser);
+          mail_gid = mkIf (cfg.mailGroup != null) (mkDefault cfg.mailGroup);
+
+          default_internal_user = mkDefault "dovecot2";
+          default_internal_group = mkDefault "dovecot2";
+
+          maildir_copy_with_hardlinks = mkDefault true;
+          pop3_uidl_format =
+            if (versionAtLeast (lib.versions.majorMinor cfg.package.version) "2.4") then
+              mkDefault "%{uidvalidity}%{uid}"
+            else
+              mkDefault "%08Xv%08Xu";
+
+          auth_mechanisms = mkDefault "plain login";
+          "service auth" = {
+            user = "root";
+          };
 
-      mailPlugins = mkIf cfg.enableQuota {
-        globally.enable = [ "quota" ];
-        perProtocol.imap.enable = [ "imap_quota" ];
-      };
+          protocols = mkDefault (
+            lib.strings.concatStringsSep " " (
+              optional cfg.enableImap "imap" ++ optional cfg.enablePop3 "pop3" ++ optional cfg.enableLmtp "lmtp"
+            )
+          );
```

Comment:
We should allow

```
protocols = ["imap" "submission" "lmtp"];
```

here so that it can be added to in a sane way. Currently I think that will expand to

```
protocols = imap
protocols = submission
protocols = lmtp
```

which might not work. Adjust it if needed so that lists of normal values get formatted in a single line, I don't think anything should need this type of formatting.